### PR TITLE
Add cycling of gems in map as well as re-centre on user's location button

### PR DESCRIPTION
**Pull request by sebastiantoh**
Other Changes:
- Removed a bunch of landmarks from being displayed on the map
- Used `vue3-google-map` instead of `@googlemaps/js-api-loader` as the latter is a bit too manual for my liking :')
- Added `vant`
- Added button to centre on current user's location (the top right hand button)
- Added buttons to cycle between dropped gems (the left and right buttons at the bottom. Note that the current middle button is a WIP. Atm, it just brings you back to your current location. But i'm not quite sure what it's supposed to do... should it be the 'drop gem button' or a button to view specific details of the gem?)

Stuff that will be improved upon in another PR
- styling and UI (any suggestions are greatly appreciated!)
- pull out css styles (i'm thinking of using `less` / `sass`. Currently leaning towards less since `vent` uses `less` as well. Not sure if you all have a preference?)

### CODE REVIEW

**Comment by ming-00** [comment]
Looks v good!! Think I need more time to read through carefully but I noticed the right button cycles through gems while the left button doesn't? Not sure if it is intended :-D UI wise, I'm a bit stuck between clicking a marker to have a tooltip pop up of gem info kinda thing vs a fixed block centered at where the WIP button is, not sure which you think is better?
I'm okay with going with less too!

**Comment by sebastiantoh** [comment]
> Looks v good!! Think I need more time to read through carefully but I noticed the right button cycles through gems while the left button doesn't? Not sure if it is intended :-D 

Oops, fixed. ty!

> UI wise, I'm a bit stuck between clicking a marker to have a tooltip pop up of gem info kinda thing vs a fixed block centered at where the WIP button is, not sure which you think is better?

hmm, I'll explore both ideas!  For the fixed block, my concern is that it'll be too narrow (on a mobile) to contain much useful info, but we can see how it goes!


> I'm okay with going with less too!

great!

**Comment by ming-00** [comment]
> hmm, I'll explore both ideas! For the fixed block, my concern is that it'll be too narrow (on a mobile) to contain much useful info, but we can see how it goes!

For block I thinkkk it should be okay bc there isn't much info to display anyway? I think block makes the cycle buttons make more sense, but it's more natural for ppl to click on the markers for a tooltip to appear imo... If tooltip then the buttons might look confusing 🤔 I think we can draft on Figma first?

File: frontend/src/components/GemMap.vue [hunk 1]
```diff
@@ -59,22 +101,105 @@ export default defineComponent({
...
+  },
+
+  methods: {
+    centerMapOnCurrentLocation() {
+      this.mapRef?.map?.panTo(this.currPosition);
+    },
+    nextGem() {
+      if (this.sortedGems.length == 0) {
+        // TODO: add better handling of this case
+        return;
```
Comment (ming-00) [inline frontend/src/components/GemMap.vue]: hmm does no buttons make sense

**Review by ming-00 — DISMISSED** [review]
Lgtm! Btw what's the line surrounding sg border?

**Comment by sebastiantoh** [comment]
> Lgtm! Btw what's the line surrounding sg border?

oh, i think that's the actual border between SG / Malaysia?

image

File: frontend/src/components/GemMap.vue (hunk 1 above)
Comment (sebastiantoh) [reply frontend/src/components/GemMap.vue]: yep, I think so! Made the changes!

**Comment by sebastiantoh** [comment]
> > hmm, I'll explore both ideas! For the fixed block, my concern is that it'll be too narrow (on a mobile) to contain much useful info, but we can see how it goes!
> 
> For block I thinkkk it should be okay bc there isn't much info to display anyway? I think block makes the cycle buttons make more sense, but it's more natural for ppl to click on the markers for a tooltip to appear imo... If tooltip then the buttons might look confusing thinking I think we can draft on Figma first?

sure!

**Comment by ming-00** [comment]
> oh, i think that's the actual border between SG / Malaysia?
> 
> image

omg LOL how did I never notice this using google maps before

**Review by ming-00 — APPROVED** [review]
lgtm!

**Comment by ming-00** [comment]
@googlemaps/js-api-loader is still needed? :0

**Comment by sebastiantoh** [comment]
i don't think so... I found this issue https://giters.com/googlemaps/js-api-loader/issues/369?amp=1 which seems related. Let me try to fix it! What version of node are you all using?

**Pull request merged**